forui/lib/src/widgets/tabs/tabs.dart

```diff
@@ -0,0 +1,168 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import 'package:flutter_localizations/flutter_localizations.dart';
+
+import 'package:forui/forui.dart';
+
+part 'tabs_style.dart';
+
+part 'tab_controller.dart';
+
+/// An object that represents a tab entry in a group of tabs.
+class FTabEntry {
+  /// The label.
+  final String? label;
+
+  /// A raw label.
+  final Widget? rawLabel;
+
+  /// The content of a tab.
+  final Widget content;
+
+  /// Creates a [FTabs].
+  FTabEntry({
+    required this.content,
+    this.label,
+    this.rawLabel,
+  }) : assert(label == null || rawLabel == null,
+            'Cannot provide both a label and a rawLabel.');
+}
+
+/// A [FTabs] that allows switching between tabs.
+class FTabs extends StatefulWidget {
+  /// The tab and it's corresponding view.
+  final List<FTabEntry> tabs;
+
+  /// The initial tab that is selected.
+  final int initialIndex;
+
+  /// Whether this tab bar can be scrolled horizontally.
+  ///
+  /// If [scrollable] is true, then each tab is as wide as needed for its label
+  /// and the entire [TabBar] is scrollable. Otherwise each tab gets an equal
+  /// share of the available space.
+  final bool scrollable;
+
+  /// The tab controller.
+  final FTabController? controller;
+
+  /// The style.
+  final FTabsStyle? style;
+
+  /// A callback that returns the tab that was tapped.
+  final ValueChanged<int>? onTap;
+
+  /// Creates a [FTabs].
+  FTabs({
+    required this.tabs,
+    this.initialIndex = 0,
+    this.scrollable = false,
+    this.controller,
+    this.style,
+    this.onTap,
+    super.key,
+  }) : assert(tabs.isNotEmpty, 'Must have at least 1 tab provided');
```

Comment:
Changing the assertion to `0 < tabs.length` should allow the `FTabs` constructor to be const.